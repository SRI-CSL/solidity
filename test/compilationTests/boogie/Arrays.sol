pragma solidity ^0.4.23;

contract Arrays {
    uint[] arr;

    function readStateArr(uint i) view public returns (uint) {
        if (i < arr.length) return arr[i];
        return 0;
    }

    function writeStateArr(uint i, uint value) public {
        arr[i] = value;
    }

    function writeParamArr(uint[] paramArr, uint i, uint value) pure public returns (uint) {
        paramArr[i] = value;
        return paramArr[i];
    }

    function readParamArr(uint[] paramArr, uint i) pure public returns (uint) {
        if (i < paramArr.length) return paramArr[i];
        return 0;
    }

    function callWithLocalArray(uint[] paramArr) pure public returns (uint) {
        return readParamArr(paramArr, 123);
    }

    function callWithStateArray() view public returns (uint) {
        return readParamArr(arr, 456);
    }
}