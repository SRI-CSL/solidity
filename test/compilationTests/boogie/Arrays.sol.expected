Verifying Arrays::readStateArr: OK
Verifying Arrays::writeStateArr: OK
Verifying Arrays::readParamArr: OK
Verifying Arrays::callWithLocalArray: OK
Verifying Arrays::callWithStateArray: OK
No errors found.
